src/SurveyService.Data/QuestionRepository.cs

```diff
@@ -11,6 +13,25 @@ public class QuestionRepository : IQuestionRepository
 {
   private readonly IDataProvider _provider;
 
+  private IQueryable<DbQuestion> CreateGetPredicates(
+    GetQuestionFilter filter,
+    IQueryable<DbQuestion> dbQuestions)
+  { 
+    if (filter.IncludeAnswers)
+    {
+      dbQuestions = dbQuestions
+        .Include(question => question.Options.Where(option => option.IsActive))
+        .ThenInclude(option => option.UsersAnswers.Where(ua => filter.IncludeAnswers));
```

Comment:
Do you need "Where" here?
(ua => filter.IncludeAnswers) true for each UserAnswer..

Reply:
Done